.github/workflows/CI-distro.yml

```
@@ -0,0 +1,108 @@
+  # Cancel intermediate builds, but only if it is a pull request build.
+  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}
+
+jobs:
+  generate-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      gap-packages: ${{ steps.set-matrix.outputs.gap-packages }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: "Set up Julia"
+        uses: julia-actions/setup-julia@v2
+        with:
+          version: '1'
+      - name: "Cache artifacts"
+        uses: julia-actions/cache@v2
+      - name: "Build package"
+        uses: julia-actions/julia-buildpkg@v1
+      - name: Get list of GAP packages
+        id: set-matrix
+        run: julia --project=. -e '
+                  using Artifacts, TOML;
+                  output = sprint(print, "gap-packages=", map(name -> name[9:end], sort!(collect(filter(startswith("GAP_pkg_"), keys(TOML.parsefile(find_artifacts_toml(Base.active_project()))))))));
+                  println(output);
+                  open(ENV["GITHUB_OUTPUT"], "a") do io;
+                    println(io, output);
+                  end;'
+
+  test:
+    name: ${{ matrix.gap-package }}
+    needs: generate-matrix
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 20
+    continue-on-error: ${{ matrix.julia-version == 'nightly' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        julia-version:
+          - '1'
+        julia-arch:
+          - x64
+        os:
+          - ubuntu-latest
+        gap-package: ${{fromJSON(needs.generate-matrix.outputs.gap-packages)}}
+        exclude:
+          - gap-package: 'alnuth'                     # `AL_EXECUTABLE, the executable for PARI/GP, has to be set`
+          #- gap-package: 'autodoc'                    # test failure
+          #- gap-package: 'caratinterface'             # `Carat program Bravais_inclusions failed with error code 1`
+          - gap-package: 'curlinterface'              # no jll
+          - gap-package: 'example'                    # no jll
+          #- gap-package: 'examplesforhomalg'          # `Error, found no GAP executable in PATH`
+          #- gap-package: 'gapdoc'                     # test failure
+          - gap-package: 'guarana'                    # `AL_EXECUTABLE, the executable for PARI/GP, has to be set`
+          - gap-package: 'hap'                        # `polymake command not found. Please set POLYMAKE_COMMAND by hand`
+          - gap-package: 'hapcryst'                   # `polymake command not found. Please set POLYMAKE_COMMAND by hand`
+          #- gap-package: 'io'                         # test failure
+          - gap-package: 'itc'                        # dependency `xgap` has no jll
+          #- gap-package: 'localizeringforhomalg'      # `Error, found no GAP executable in PATH`
+          #- gap-package: 'normalizinterface'          # `Error, Some error in the normaliz input data detected: Cone Property in last line not allowed for lattice ideals`
```

```suggestion
          - gap-package: 'normalizinterface'          # Tests fail: NormalizInterface currently bundles Normaliz 3.9.3 and its test suite is tuned to that, but our JLL builds it against normaliz_jll which has 3.10.2
```

This could be fixed by an update to NormalizInterface